common/ledger/blkstorage/block_stream.go

```diff
@@ -105,8 +105,8 @@ func (s *blockfileStream) nextBlockBytesAndPlacementInfo() ([]byte, *blockPlacem
 	if lenBytes, err = s.reader.Peek(peekBytes); err != nil {
 		return nil, nil, errors.Wrapf(err, "error peeking [%d] bytes from block file", peekBytes)
 	}
-	length, n := proto.DecodeVarint(lenBytes)
-	if n == 0 {
+	length, n := protowire.ConsumeVarint(lenBytes)
+	if n <= 0 {
```

Comment:
what is the use case for negative value of n?

Reply:
The DecodeVarint method checks for a negative value of n. I just repeated it.

https://github.com/golang/protobuf/blob/master/proto/buffer.go#L41

Reply:
ConsumeVarint can return -1